content/guides/bake/index.md

```diff
@@ -0,0 +1,499 @@
+├─ linux/riscv64        0eae8ba0367a       9.18MB         9.18MB
+├─ linux/arm64          56561051c49a         30MB         9.89MB
+└─ linux/amd64          e8ca65079c1f        9.8MB          9.8MB
+
+bakeme:latest           20065d2c4d22       44.4MB         25.9MB
+├─ linux/riscv64        7cc82872695f       8.21MB         8.21MB
+├─ linux/arm64          e42220c2b7a3       27.1MB         8.93MB
+└─ linux/amd64          af5b2dd64fde       8.78MB         8.78MB
+```
+
+## Exporting build artifacts
+
+Exporting build artifacts like binaries can be useful for deploying to
+environments without Docker or Kubernetes. For example, if your programs are
+meant to be run on user's local machine.
+
+With programming languages like Go and Rust where the compiled binaries are
+usually portable, creating alternate build targets for exporting only the
+binary is trivial. All you need to do is add an empty stage in the Dockerfile
+containing only the binaries
+
+In the `docker-bake.hcl` file, create a new `binaries` stage similar to the
+`default` stage that builds images. You can omit the `tags` and `attest`
+attributes since they're only relevant for Docker images. Instead, this stage
+should set the `output` attribute to export the results to the filesystem.
+
+```hcl
+target "binaries" {
+  matrix = {
+    mode = ["release", "debug"]
+  }
+  name = "binaries-${mode}"
+  target = "binaries"
+  args = {
+    BUILD_TAGS = mode
+  }
+  output = ["type=local,dest=./build/${mode}"]
+  platforms = [
+    "linux/amd64",
+    "linux/arm64",
+    "linux/riscv64",
+  ]
+}
+```
+
+This target assumes that there's a `binaries` stage in the Dockerfile that
+contains the binaries that you want to export. The contents of that stage are
+exported to the local filesystem path `./build/${mode}/`, where `mode` is the
+expanded matrix variable, e.g. `./build/debug/`.
+
+Next, add the `binaries` stage to the Dockerfile:
+
+```dockerfile
+FROM scratch AS binaries
+COPY --from=build "/usr/bin/bakeme" /
+```
+
+Now, running `docker buildx bake binaries` builds all of the variants of your
+program and exports them to the local `build/` directory. Subdirectories are
+automatically created for platform variants.
+
+```console
+$ docker buildx bake binaries
+$ tree ./build
+./build
+├── debug
+│   ├── linux_amd64
+│   │   └── bakeme
+│   ├── linux_arm64
+│   │   └── bakeme
+│   └── linux_riscv64
+│       └── bakeme
+└── release
+    ├── linux_amd64
+    │   └── bakeme
+    ├── linux_arm64
+    │   └── bakeme
+    └── linux_riscv64
+        └── bakeme
+
+9 directories, 6 files
+```
+
+Generating all the variants at once is nice for release purposes, but for local
+development and testing, maybe you just want a quick way to build a single
+binary for your local platform. Buildx has a special `local` platform value for
+that.
+
+Add another Bake target for generating and exporting only the local platform
+binary:
+
+```hcl
+target "localbin" {
```

Comment:
I think binaries should be `local` platform by default like we do in buildx: https://github.com/docker/buildx/blob/1de332530f6c21e69ddf8577fbc7adcf0ae7c723/docker-bake.hcl#L145-L150 and if people want to "cross" they can just inherit from it instead: https://github.com/docker/buildx/blob/1de332530f6c21e69ddf8577fbc7adcf0ae7c723/docker-bake.hcl#L152-L167